apollo-federation/src/sources/connect/variable.rs

```diff
@@ -144,24 +145,25 @@ impl Display for Namespace {
 /// A variable reference. Consists of a namespace starting with a `$` and an optional path
 /// separated by '.' characters.
 #[derive(Debug, Eq, PartialEq, Clone, Hash)]
-pub struct VariableReference<'a, N: FromStr + ToString> {
+pub struct VariableReference<N: FromStr + ToString> {
     /// The namespace of the variable - `$this`, `$args`, `$status`, etc.
     pub namespace: VariableNamespace<N>,
 
     /// The path elements of this reference. For example, the reference `$this.a.b.c`
     /// has path elements `a`, `b`, `c`. May be empty in some cases, as in the reference `$status`.
-    pub(crate) path: Vec<VariablePathPart<'a>>,
+    pub(crate) selection: SelectionTrie,
 
     /// The location of the reference within the original text.
-    pub(crate) location: Range<usize>,
+    pub(crate) location: Option<Range<usize>>,
 }
 
-impl<N: FromStr + ToString> Display for VariableReference<'_, N> {
+impl<N: FromStr + ToString> Display for VariableReference<N> {
     fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
         f.write_str(self.namespace.namespace.to_string().as_str())?;
-        for part in &self.path {
-            f.write_str(".")?;
-            f.write_str(part.as_str())?;
+        if !self.selection.is_empty() {
```

Comment:
`write!(f, " {{ {} }} ", self.selection.to_string().as_str())` should work as well

Reply:
https://github.com/apollographql/router/pull/7419/commits/6bf390b54abe2d4d7e5c9ab6556d54f0b25035d3